model_fitting/train_random_forest.py

```diff
@@ -0,0 +1,221 @@
+import json 
+import sys
+import seaborn as sns
+import joblib
+import os
+import matplotlib.pyplot as plt
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import cross_val_score, StratifiedKFold, cross_validate
+from sklearn.metrics import plot_roc_curve
+import pandas as pd
+import numpy as np
+
+def parse_data(file_path):
+    try:
+        data = pd.read_csv(file_path, engine='python')
+    except Exception as e:
+        exit(f'Cannot analyse data! {e}')
+    
+    # create masks for separating train and test samples (fifth sample of each mAb)    
+    test_rows_mask = data['sample_name'].str.contains('test')
+    train_rows_mask = ~test_rows_mask
+
+    # separating train and test samples (fifth sample of each mAb)
+    # a matrix of the actual feature values
+    y_train = data['label'][train_rows_mask]
+    y_test = data['label'][test_rows_mask]
+
+    sample_names_train = data['sample_name'][train_rows_mask]
+    sample_names_test = data['sample_name'][test_rows_mask]
+
+    # don't need the labels and sample names anymore
+    data.drop(['sample_name', 'label'], axis=1, inplace=True)
+    # a matrix of the actual feature values
+    X_train = data[train_rows_mask].values
+    X_test = data[test_rows_mask].values
+
+    feature_names = np.array(data.columns)
+
+    return X_train, y_train, X_test, y_test, feature_names, sample_names_train, sample_names_test
+
+
+def generate_roc_curve(X, y, classifier, number_of_features, output_path):
+    ax = plt.gca()
+    plot_roc_curve(classifier, X, y, **{'marker': 'o'}, ax=ax)
+    plt.savefig(f"{output_path}/roc_curve_{number_of_features}.png", format='png', bbox_inches="tight")
+    plt.close()
+
+
+def generate_heat_map(df, number_of_features, hits_data, number_of_samples, output_path):
+    train_data = np.log2(df+1) if hits_data else df 
```

Comment:
Same comment as above, this doesn't support p-value correctly

Reply:
I thought it was decided after we showed to the client the logs to leave only for hits.